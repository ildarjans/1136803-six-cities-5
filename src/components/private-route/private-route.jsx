import React from "react";
import {Route, Redirect} from "react-router-dom";
import PropTypes from "prop-types";
import {connect} from "react-redux";
import {AppRoute, AuthorizationStatus} from "../../const";

export const PrivateRouteComponent = (props) => {
  const {authorizationStatus, path, isExact, render} = props;
  return (
    <Route
      exact={isExact}
      path={path}
      render={() => {
        return (
          authorizationStatus === AuthorizationStatus.AUTH ?
            render() :
            <Redirect to={AppRoute.LOGIN}/>
        );
      }}
    />
  );
};

PrivateRouteComponent.propTypes = {
  authorizationStatus: PropTypes.string.isRequired,
  path: PropTypes.string.isRequired,
  isExact: PropTypes.bool.isRequired,
  render: PropTypes.func.isRequired,
};

const mapStateToProps = ({USER}) => ({
  authorizationStatus: USER.authorizationStatus
});

export const PrivateRoute = connect(mapStateToProps)(PrivateRouteComponent);
