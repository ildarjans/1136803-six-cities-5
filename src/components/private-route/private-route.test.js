import React from "react";
import renderer from "react-test-renderer";
import {BrowserRouter} from "react-router-dom";
import {PrivateRoute} from "./private-route";

const isAuthStatus = `AUTHORIZED`;
const isNotAuthStatus = `NO_AUTHORIZED`;
const noop = () => {};

describe(`Should PrivateRoute rendered correctly`, () => {
  it(`is Authorized`, () => {
    const tree = renderer.create(
        <BrowserRouter>
          <PrivateRoute
            authorizationStatus={isAuthStatus}
            path={`/`}
            isExact={true}
            render={noop}
          />
        </BrowserRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it(`is not Authorized`, () => {
    const tree = renderer.create(
        <BrowserRouter>
          <PrivateRoute
            authorizationStatus={isNotAuthStatus}
            path={`/`}
            isExact={true}
            render={noop}
          />
        </BrowserRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
