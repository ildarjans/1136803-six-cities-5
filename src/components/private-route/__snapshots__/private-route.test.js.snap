// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should PrivateRoute rendered correctly is Authorized 1`] = `null`;

exports[`Should PrivateRoute rendered correctly is not Authorized 1`] = `null`;
